sdks/java/io/cassandra/src/main/java/org/apache/beam/sdk/io/cassandra/CassandraIO.java

```diff
@@ -373,7 +419,86 @@ private CassandraIO() {}
       checkArgument(entity() != null, "withEntity() is required");
       checkArgument(coder() != null, "withCoder() is required");
 
-      return input.apply(org.apache.beam.sdk.io.Read.from(new CassandraSource<>(this, null)));
+      ReadAll<T> readAll = CassandraIO.readAll();
+
+      return input
+          .apply(Create.of(this))
+          .apply("Split", ParDo.of(new SplitFn()))
```

Comment:
Can you please move this Split into the ReadAll expand method. When not specified RingRanges like in the `read()` case we should split them on the `ReadAll` expansion. Of course if RingRanges are specified we will probably ignore recalculating them.

Reply:
I don't think it makes sense to move to the readAll...  The readAll takes a Read<>, so on encountering each Read<>, we want to check if there isn't a ringRange or Query set, and then expand it?  I think it is much cleaner here, and I think this is where you originally suggested the split happen?